### ISSUE TITLE
Create flavour for targetsdk 29

### ISSUE THREAD

**Issue by licaon-kter** [syncthing/syncthing-android#1746]
Given other Android stores forced limitations, Syncthing can't access files created by user apps: https://github.com/syncthing/syncthing-android/issues/1638

Since this app is available in other Android stores, that are not run by monopolies funded by user data selling that force users to upload app data onto their servers unencrypted via their root enabled system apps, suddenly Syncthing becomes unable to fullfill its functions.

Slowly they'll be nothing to sync anymore as Syncthing will not be able to access any data at all.

Eg. Now it can't backup OSMand~ favourite places, can't backup DevilutionX game saves, etc.

Can targetsdk be separated in gradle flavours?

If not, it's ok for F-Droid to downgrade targetsdk in its builds?

**Comment by imsodin** [comment]
I understand these limitations are annoying. And believe me, I'd love if there was a well-adobted (adobtable) platform for open android apps. I personally do use microG on lineageos with F-Droid as my main app store. However that's just not an accessible option for most users, which are locked into the Google ecosystem. And I do want those users to be able to use Syncthing.

If anyone wants to maintain something like this, lets talk about it and see if it's feasible. My current understand is, that it's not. And target SDK is just one piece of the puzzle, the restrictions to `Android/data` is also a Android 11 restrictions (regardless of target SDK). And then completely orthogonally there's the advantage of now being able to access SD card content, which is also a big deal.  

tl;dr: Until someone steps-up to co-maintain, this is an unconditional no.

PS: That also includes targeting 29 on F-Droid. That might work, but it's not tested, thus I don't want it to happen under the official name.

**Comment by licaon-kter** [comment]
Rather unfortunate then, sadly this forces non-Google users to either fork the app or switch to an app that's not limited by other stores policies (does one exist?) or reinstall an older version before the user hostile changes were introduced _(so much for "security reasons" thanks Google)_.

I'll see what a fork entails, its within FOSS power, yet it should be truly the last solution...

/PS: I rather loath forks usually since they're spawn out of selfish reasons, eg. Change app name or colour, but here it feels right, albeit my skills are limited... so the progress/future of such a fork will be limited too. :(

**Comment by imsodin** [comment]
1. I might have communicated badly: I am open to someone (e.g. you) doing this work within this app/project. I just want someone to commit to maintaining that aspect, which basically means co-maintaining this app. That includes review and testing (e.g. I'd need to see evidence that targeting 29 does anything on android >= 11 - as far as I understand you still can't access `Android/data` then). Technically if this can be handled in build flavors, keeping both the old and new methods to request storage permissions doesn't seem like an undue burden. From my point of view the main tasks are testing and the build setup.
2. If there's indeed no way to access `Android/data` on Android >= 11, there might a point in having a "syncthing-android-for-android-10-and-below". I'd expect maintaining a few patches on top of this project in a fork and publishing that under some name like that (obviously not exactly that one :) ) to be feasible. I wouldn't personally consider that worthwhile, but that's not what is in question here.
3. There is already an active fork: https://github.com/catfriend1/syncthing-android

Basically if you are interested in co-maintaining Syncthing, lets talk about that here. Otherwise please move further discussion to the appropriate venue, e.g. https://forum.syncthing.net.

**Comment by licaon-kter** [comment]
Of course, this issue is just a reaction to "it can't be done!!!111" posts of https://github.com/syncthing/syncthing-android/issues/1638

Sadly showing that _other apps_ actually can yielded no result. Yes, that needs code...people, so open that one so it's clear you want this fixed...not just settle on "no way".

**Comment by AudriusButkevicius** [comment]
Also, don't forget that syncthing is written in go.

So in my book it's still "can't be done", because it can't be done without an obscene amount of effort.

Unless you plan on working on this, I say it stays closed, as it's unlikely to get fixed.

**Comment by licaon-kter** [comment]
Over the years I think I saw "syncthing is written in go" as an excuse written multiple times, yet the app works fine still . :)

**Comment by imsodin** [comment]
That's basically it. I am not interested in having discussions without first clarifying the aim. Which might be (purely speculating here): You, @licaon-kter, have an idea how to get access to #1638 with an alternative build flavor, and are interested in working and maintaining an implementation. Then I am interested in spending time to discuss this. I haven't yet seen any such statement. First step to keep me in the loop is stop talking about "excuses" and actually engaging with the questions and points raised from others (e.g. your "app X does Y statements" lack any info on android version).
